cobc/tree.c

```diff
@@ -6174,8 +6174,11 @@ cb_build_binary_op (cb_tree x, const enum cb_binary_op_op op, cb_tree y)
 		/*
 		 * If this is an operation between two literal strings
 		 * then resolve the value here at compile time -> "constant folding"
+		 *
+		 * TODO: build cob_fields and call cob_cmp from libcob.
 		 */
 		} else if (cb_constant_folding
+		 && current_program->collating_sequence == NULL
```

Comment:
it seems that the check for = and ~ below can still be done - and in the other cases the literal's class should likely be checked against national in which case `collating_sequence_n` must be compared

Reply:
The new version of this code is fine!